tools/kwiver_tool_runner.cxx

```diff
@@ -186,21 +333,59 @@ int main(int argc, char *argv[])
 
   // initialize the global context
   tool_context->m_wtb.set_indent_string( "      " );
-
+  cmd_root = build_command_tree( vpm );
   command_line_parser options( argc, &argv );
 
-  if ( (options.m_applet_name == "help") || (argc == 1) )
+  // If there is only a 'help' as an argument
+  if ( (argc == 1) || (options.m_applet_name[0] == "help") )
   {
-    help_applet( options, tool_context, vpm );
+    tool_runner_usage( tool_context, vpm );
     return 0;
   } // end help code
 
   // ----------------------------------------------------------------------------
   try
   {
     // Create applet based on the name provided
-    applet_factory app_fact;
-    kwiver::tools::kwiver_applet_sptr applet( app_fact.create( options.m_applet_name ) );
+    auto cn = cmd_root;
+    size_t ci;
+    for ( ci = 0; ci < options.m_applet_name.size(); ci++)
+    {
+      auto n = cn->find( options.m_applet_name[ci] );
+      if ( ! n )
+      {
+        break;
+      }
+      cn = n;
+    } // end for
+
+    // If there are no matches or we run out of command words
+    // before we reach a terminal command node.
+    if ( ci == 0 && !cn->is_command() )
```

Comment:
Is there ever a case where `ci == 0` and `cn` is a command?  Doesn't `ci == 0` imply that `cn == cmd_root`?  I suppose it can't hurt to check.

Reply:
Commands can be present at the root if it is a one word command. This is the case when supporting legacy plugins.

Reply:
Understood, but isn't the root always the node named "Root" that contains everything else and is not a command?  So those top-level commands would be at `ci == 1`, right?

Reply:
This can happen if the command line is malformed and there are no elements in the `m_applet_name` vector.